File: scripts/python/sacred_cli/button.py
"""Terminal buttons. Mirrors scripts/cli/lib/button.js."""

from .ansi import COLORS, RESET, RESET_FG, bg_hex, fg_hex, vis_len


def button(hotkey: str, label: str) -> str:
    return (
        f"{bg_hex(COLORS['btnHotkey'])}{fg_hex(COLORS['text'])} {hotkey} {RESET_FG}"
        f"{bg_hex(COLORS['btnLabel'])}{fg_hex(COLORS['btnLabelText'])} {label.upper()} {RESET}"
    )


_bg_win = bg_hex(COLORS["windowBg"])


def button_row(left: str, right: str, inner_w: int) -> str:
    gap = max(1, inner_w - vis_len(left) - vis_len(right))
    return f"{left}{_bg_win}{' ' * gap}{RESET}{right}"

File: scripts/cli/lib/button.ts
import { bgHex, fgHex, RESET, RESET_FG, COLORS, visLen } from './ansi';

function button(hotkey: string, label: string): string {
  return `${bgHex(COLORS.btnHotkey)}${fgHex(COLORS.text)} ${hotkey} ${RESET_FG}${bgHex(COLORS.btnLabel)}${fgHex(COLORS.btnLabelText)} ${label.toUpperCase()} ${RESET}`;
}

//NOTE(jimmylee): Gap explicitly uses windowBg so RESET in buttons does not leak terminal default bg.
const bgWin: string = bgHex(COLORS.windowBg);

function buttonRow(left: string, right: string, innerW: number): string {
  const gap = Math.max(1, innerW - visLen(left) - visLen(right));
  return `${left}${bgWin}${' '.repeat(gap)}${RESET}${right}`;
}

export { button, buttonRow };
